datashader/datatypes.py

```diff
@@ -635,8 +635,10 @@ def astype(self, dtype, copy=True):
         elif is_extension_array_dtype(dtype):
             return dtype.construct_array_type()._from_sequence(
                 np.asarray(self))
-
-        return np.array([v for v in self], dtype=dtype, copy=copy)
+        if copy:
+            return np.array([v for v in self], dtype=dtype)
+        else:
+            return np.asarray([v for v in self], dtype=dtype)
```

Comment:
Little confused by this (and the old code too), passing a list to an array constructor by definition makes a copy. So I'd simply do this:

```suggestion
            return np.array([v for v in self], dtype=dtype)
```

Reply:
Should we have it so if `copy` is false we don't use a list comprehension?

Reply:
Good question, I'm a little unclear on why we use a list comprehension in the first place.